### PR TITLE
#604 eslint-plugin-vue added, code adopted to Vue Style Guide

### PR DESCRIPTION
**Pull request by PanMisza**

### CODE REVIEW

**Comment by pkarw** [comment]
Thanks! Huge amount of work here! @filrak @mercs600 @szafran89 guys may I ask You for some CR here?

**Comment by filrak** [comment]
@PanMisza have you tested it after changes?

**Comment by PanMisza** [comment]
@filrak 
I think that everything works after changes, but I don't know this project as well as you so it's possible that I'm not checked everything. I will be grateful if you make sure that these changes have not broken something.

Problems can appear with props - in places where I could predict what type to define, I defined it. But there are some components where I just set type to `null` (so it accepts any type). Also not in every place I'm sure that props is required or not. But even when prop type is not valid, Vue just produce console warning and thing should work anyway.

**Comment by pkarw** [comment]
@PanMisza please also resolve the conflicts which master;
I’ve started the review but will be able to finish it later today; generally I would only suggest some validation changes and check if the logic isn’t broken by the changes - at first glance it seems like no major issues introduced by this refactor

File: src/components/core/SizeButton.vue [hunk 1]
```diff
@@ -15,7 +15,28 @@
...
+    },
+    code: {
+      type: null,
+      required: false,
+      default: () => false
+    },
+    context: {
+      type: null,
+      required: false,
+      default: () => false
```
[the remark this answers is not in the document]
Comment (mercs600) [reply src/components/core/SizeButton.vue]: @Aekal not in this case. When you look on documentation, you can see that properly default value for object is function.

[the remark this answers is not in the document]
Comment (mercs600) [reply src/components/core/SizeButton.vue]: https://vuejs.org/v2/guide/components.html#Prop-Validation

[the remark this answers is not in the document]
Comment (Aekal) [reply src/components/core/SizeButton.vue]: Ah ok, it makes sense :) I didn't see it before, sorry then

**Comment by filrak** [comment]
@PanMisza Fix conflicts so i can merge it ;) (And discuss the returns with @Aekal please ;) )

Great job

**Comment by pkarw** [comment]
@filrak if you think its ready, please merge it in; from my side it looks fine 
@PanMisza - probably you’re just conflicts to be solved from first accepted PR; quite a huge PR :) fingers crossed

File: src/store/modules/cart.js [hunk 2]
```diff
@@ -289,7 +289,7 @@ const store = {
           silent: true,
           callback_event: 'servercart-after-itemdeleted'
         }, { root: true }).then(task => {
-          return
+
```
[the remark this answers is not in the document]
Comment (PanMisza) [reply src/store/modules/cart.js]: OK, I'll keep this. Thanks!

File: src/themes/catalog/components/core/SizeFilter.vue [hunk 3]
```diff
@@ -23,7 +23,6 @@ export default {
         if (filterOption.id === this.id) {
           if (this.active) {
             this.active = false
-            return
```
[the remark this answers is not in the document]
Comment (PanMisza) [reply src/themes/catalog/components/core/SizeFilter.vue]: Why?

File: src/components/core/PriceButton.vue [hunk 4]
```diff
@@ -6,7 +6,32 @@
 
 <script>
 export default {
-  name: 'price-button',
-  props: ['content', 'id', 'code', 'from', 'to', 'context']
+  name: 'PriceButton',
```
[the remark this answers is not in the document]
Comment (PanMisza) [reply src/components/core/PriceButton.vue]: Thanks! I will change it :)

**Comment by PanMisza** [comment]
@filrak @pkarw 
Conflicts resolved.

**Comment by filrak** [comment]
@PanMisza thank you for your work ;)

**Pull request merged**